pkg/rancher-desktop/backend/lima.ts

```diff
@@ -1557,18 +1558,46 @@ export default class LimaBackend extends events.EventEmitter implements VMBacken
     return JSON.parse(stdout).SPNetworkDataType;
   }
 
+  protected async installContainerdShims() {
+    // shims bundled with the app itself
+    const bundleDir = path.join(paths.resources, 'linux', 'internal');
+    // shims managed by the user
+    const shimDir = path.join(paths.appHome, 'containerd-shims');
+
+    await this.execCommand('mkdir', '-p', 'bin');
+    await this.writeFile('bin/install-containerd-shims', INSTALL_CONTAINERD_SHIMS_SCRIPT, 'a+x');
+    await this.execCommand({ root: true }, 'bin/install-containerd-shims', bundleDir, shimDir);
+  }
+
   protected async configureContainerd(): Promise<void> {
     const workdir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-containerd-install-'));
 
     try {
+      const configureWASM = !!this.cfg?.experimental?.containerEngine?.webAssembly?.enabled;
+
       await this.writeFile('/usr/local/bin/nerdctl', NERDCTL, 0o755);
 
       await this.execCommand({ root: true }, 'mkdir', '-p', '/etc/cni/net.d');
-
       if (this.cfg?.kubernetes.options.flannel) {
         await this.writeFile('/etc/cni/net.d/10-flannel.conflist', FLANNEL_CONFLIST);
       }
-      await this.writeFile('/etc/containerd/config.toml', CONTAINERD_CONFIG);
+
+      await this.writeFile('/etc/containerd/config.toml', await BackendHelper.containerdTOML(configureWASM));
+
+      // Use containerd-snapshotter if wasm support is enabled
+      if (this.cfg?.containerEngine.name === ContainerEngine.MOBY) {
+        let daemonConfig: Record<string, any>;
+
+        try {
+          daemonConfig = JSON.parse(await this.execCommand({ capture: true, root: true }, 'cat', DOCKER_DAEMON_PATH));
```

Comment:
There's a [`readFile()`](https://github.com/rancher-sandbox/rancher-desktop/blob/962b55e65013d73c2cba883b0dbe0df4d80d9bf9/pkg/rancher-desktop/backend/lima.ts#L1583) that may make the intention clearer. (It also just runs `cat`.)

Reply:
Ok, will change; missed that.